### PR TITLE
Error Handling for publish failure from mqttcm when component is crashed/not ready

### PR DESCRIPTION
**Pull request by nlrcomcast**

### CODE REVIEW

File: src/mqttcm_webcfg.c [hunk 1]
```diff
@@ -120,7 +122,90 @@ void sendRbusEventWebcfgOnPublish(int mid)
 			MqttCMError("provider: rbusEvent_Publish onpublish event failed: %d\n", rc);
 	}
 }
+
+void sendRbusErrorto_mqtt(rbusError_t rc, char *topic_name)
```
Comment (guruchandru) [inline src/mqttcm_webcfg.c:126]: change this to "sendRbusErrorToMqtt" inorder to follow the same kind of function name we use in this repo

File: src/mqttcm_webcfg.c [hunk 2]
```diff
@@ -120,7 +122,90 @@ void sendRbusEventWebcfgOnPublish(int mid)
...
+	const char *module = getComponentFromTopicName(topic_name);
+	Get_Mqtt_LocationId(locationID);
+	if( Get_Mqtt_ClientId() != NULL && strlen(Get_Mqtt_ClientId()) !=0 )
+	{
+	      strncpy(g_ClientID, Get_Mqtt_ClientId(), sizeof(g_ClientID)-1);
+	      MqttCMDebug("g_ClientID fetched from Get_Mqtt_ClientId is %s\n", g_ClientID);
+	}
+	snprintf(topic_str, MAX_MQTT_LEN, "%s%s/%s/%s/poke", MQTT_PUBLISH_NOTIFY_TOPIC_PREFIX, g_ClientID,locationID,module);
 	
+	payload = createcJsonSchema(rc,topic_name);
```
Comment (guruchandru) [inline src/mqttcm_webcfg.c:142]: add NULL check for topic name

File: src/mqttcm_webcfg.c (hunk 1 above)
Comment (nlrcomcast) [reply src/mqttcm_webcfg.c]: sendRbusErrorto_mqtt function name changed to sendRbusErrorToMqtt

File: src/mqttcm_webcfg.c (hunk 2 above)
Comment (nlrcomcast) [reply src/mqttcm_webcfg.c:147]: NULL check added for topic_name

**Review by harnishypatel — APPROVED** [review]

**Pull request merged**